### PR TITLE
Update dependencies next phase 2

### PR DESCRIPTION
**Pull request by awharn**
**What It Does**

Updates the remaining packages (except `yargs` and `typescript`) to their latest versions
Replaces `lerna` with `lerna-lite` for a lightweight Lerna experience
Removes `get-stream` for a native node function
Removes unneeded dependencies:
- `@types/find-up`
- `shebang-regex`
- `syncpack`
- `tsc-watch`

Increases node engine required version to `>=16.7.0`

The following packages could not be fully upgraded:
`chalk` - ESM
`diff2html` - Using specialized version
`find-up` - ESM
`markdown-it` - Breaks HTML links
`string-width` - ESM
`strip-ansi` - ESM
`ts-node` - Breaks imports of `@zowe/secrets-for-zowe-sdk`
`wrap-ansi` - ESM

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: packages/imperative/src/utilities/src/CliUtils.ts [hunk 1]
```diff
@@ -484,26 +484,15 @@ export class CliUtils {
...
-                terminal: true,
-                prompt: message,
-                silent: opts?.hideText,
-                replace: opts?.maskChar,
-                timeout: secToWait ? (secToWait * 1000) : null  // eslint-disable-line @typescript-eslint/no-magic-numbers
-            }, (error: any, result: string) => {
-                if (error == null) {
-                    resolve(result);
-                } else if (error.message === "canceled") {
-                    process.exit(2);
```
Comment (t1m0thyj) [inline packages/imperative/src/utilities/src/CliUtils.ts:500]: Do we need to await the promise and wrap it in a try-catch block? The `process.exit` may be important in order for Ctrl+C to be handled properly when using daemon mode.

File: __tests__/__packages__/cli-test-utils/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe CLI test utils package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Updated dependencies for technical currency
```
Comment (anaxceron) [inline __tests__/__packages__/cli-test-utils/CHANGELOG.md:7]: End w/ a period and add a PR or issue link please.

File: packages/cli/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to the Zowe CLI package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Updated additional dependencies for technical currency
+- BugFix: Updated engine to Node 16.7.0
```
Comment (anaxceron) [inline packages/cli/CHANGELOG.md:8]: For both: End w/ a period and add a PR or issue link please.

Apply this to any changelog items that would be published in the Zowe Docs release notes.

**Review by anaxceron — APPROVED** [review]
Looks good. Just a couple of notes.

**Review by gejohnston — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @awharn!

File: package.json [hunk 4]
```diff
@@ -62,17 +67,14 @@
...
     "mustache": "^4.2.0",
     "npm-lockfile": "^3.0.7",
     "rimraf": "^5.0.0",
-    "shebang-regex": "^2.0.0",
     "symlink-dir": "^5.2.1",
-    "syncpack": "^5.6.10",
     "ts-jest": "^29.1.2",
     "ts-node": "^7.0.1",
-    "typedoc": "^0.23.20",
+    "typedoc": "^0.25.8",
```
Comment (zFernand0) [inline package.json:77]: didn't they changed the output format and if we want to keep the old format we have to use the `--categorizeByGroup` option?

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Comment by awharn** [comment]
I don't think fixing the typedoc requires a changelog entry. 😅

File: package.json (hunk 4 above)
Comment (awharn) [reply package.json:77]: Discussed at my desk, the new format looks better.

**Pull request merged**